src/main/org/pantsbuild/jarjar/JarJarMojo.java

```diff
@@ -43,7 +44,8 @@ public void execute() throws MojoExecutionException {
                 patterns = RulesFileParser.parse(rulesFile);
             }
             // TODO: refactor with Main.java
-            MainProcessor proc = new MainProcessor(patterns, verbose, true);
+            MainProcessor proc = new MainProcessor(patterns, verbose, true,
```

Comment:
this entry point is called by an external tool (maven) and you are introducing a  binary incompatibility.  We aren't adding any new switches for maven to influence this choice, so its going to break the other tool for no good reason.

I think it would be best to just leave this constructor alone and create an overload in MainProcessor.java.